contracts/extensions/token/erc20/UnitToken.sol

```diff
@@ -0,0 +1,237 @@
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ */
+
+/**
+ * 
+ The UnitTokenExtension is a contract to give erc20 functionality
+ to the internal token UNITS held by DAO members inside the DAO itself. 
+
+  */
+contract UnitTokenExtension is
+    DaoConstants,
+    AdapterGuard,
+    IExtension,
+    IERC20
+    //AccessControlEnumerable,
+{
+    //Dao address
+    DaoRegistry public dao;
+    //Pausable role to prevent transfers
+    bytes32 public constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
+    bool public initialized = false;
+
+    mapping(address => mapping(address => uint256)) private _allowances;
+
+    /// @notice Clonable contract must have an empty constructor
+    constructor() {}
+
+    /**
+     * @notice Initializes the extension with the DAO and Bank address that it belongs to.
+     * @param _dao The address of the DAO that owns the extension.
+     * @param creator The owner of the DAO and Extension that is also a member of the DAO.
+     */
+    function initialize(DaoRegistry _dao, address creator) external override {
+        require(!initialized, "already initialized");
+        require(_dao.isMember(creator), "not a member");
+        initialized = true;
+        dao = _dao;
+    }
+
+    /**
+     * @dev Returns the name of the token.
+     */
+    function name() public view virtual returns (string memory) {
+        return "Unit Token";
+    }
+
+    /**
+     * @dev Returns the symbol of the token, usually a shorter version of the
+     * name.
+     */
+    function symbol() public view virtual returns (string memory) {
+        return "UNIT";
+    }
+
+    /**
+     * @dev Returns the number of decimals used to get its user representation.
+     * For example, if `decimals` equals `2`, a balance of `505` tokens should
+     * be displayed to a user as `5,05` (`505 / 10 ** 2`).
+     */
+    function decimals() public view virtual returns (uint8) {
+        return 18;
+    }
+
+    /**
+     * @dev Returns the amount of tokens in existence.
+     */
+    function totalSupply() public view override returns (uint256) {
+        BankExtension bank = BankExtension(dao.getExtensionAddress(BANK));
+        return bank.balanceOf(TOTAL, UNITS);
```

Comment:
instead of making it work only with UNITS, I would make it a constructor param to set it for UNITS or any other internal token

Reply:
TODO @fforbeck